ui/packages/catalog-ui-search/src/main/webapp/component/visualization/maps/cesium/map.cesium.js

```diff
@@ -475,11 +475,86 @@ module.exports = function CesiumMap(
         }
       })
     },
+    /*
+     * Calculates the distance (in meters) between the two positions in the given array of
+     * Cartesian3 points.
+     */
+    calculateDistanceBetweenPositions(cartesians) {
+      const startCartographic = new Cesium.Cartographic.fromCartesian(cartesians[0])
+      const endCartographic = new Cesium.Cartographic.fromCartesian(cartesians[1])
+      const ellipsoidGeodesic = new Cesium.EllipsoidGeodesic(startCartographic, endCartographic)
+      
+      return ellipsoidGeodesic.surfaceDistance
+    },
+    /*
+     * Draws a marker on the map designating a start/end point for the ruler measurement. The given
+     * coordinates should be an object with 'lat' and 'lon' keys with degrees values. The given
+     * marker label should be a single character or digit that is displayed on the map marker.
+     */
+    addRulerPoint(coordinates, markerLabel) {
+      const { lat, lon } = coordinates
+      // a point requires an altitude value so just use 0
+      const point = [lon, lat, 0]
+      const options = {
+        id: markerLabel,
+        title: `Selected ruler coordinate '${markerLabel}'`,
+        color: '#FC2803',
+        icon: null,
+        view: this,
+      }
+
+      // calls addPointWithText() with the flag set to use the custom marker label
```

Comment:
code line below is self explanatory without the comment, although we could pass a boolean called `useCustomText` to keep things clear

Reply:
True, I might have went a bit excessive with the comments since there's not much existing documentation 😅. But sure, I can add a variable to make that method call more clear.